pypeman/plugins/base.py

```diff
@@ -0,0 +1,137 @@
+"""
+Base class for a pypeman plugin
+
+All interfaces are specified nothing is implemented
+"""
+
+import logging
+
+from inspect import iscoroutinefunction
+
+
+logger = logging.getLogger(__name__)
+
+
+class BasePlugin:
+    (INITIALIZED, READY, STARTING, STARTED, STOPPING, STOPPED, DESTROYED) = \
+        states = range(7)
+
+    STATE_NAMES = (
+        "INITIALIZED", "READY", "STARTING", "STARTED",
+        "STOPPING", "STOPPED", "DESTROYED",
+        )
+
+    def __init__(self):
+        """
+        Early initialisation before any pypeman endpoints and channels are created
+        """
+        self.name = __name__ + str(self.__class__)
```

Comment:
Use fstring here ?

Reply:
true. we abandoned py 3.5, so we can now use fstrings. Will do

Reply:
done a2bf086ab9c7b147f1de65cbf20ce21710ae8681